manager/api/config/src/main/java/io/apiman/manager/api/config/Version.java

```diff
@@ -42,32 +41,29 @@
      */
     public Version() {
     }
-    
-    @PostConstruct
-    public void postConstruct() {
-        load();
-    }
```

Comment:
What prompted this change?

Reply:
The problem was in `manager/api/war/tomcat8/src/main/java/io/apiman/manager/api/war/tomcat8/Tomcat8ApiManagerApplication.java` in this line:
```java
 beanConfig.setVersion(new Version().getVersionString());
```
The result of `new Version().getVersionString()` was always null because the postConstruct was somehow not called.
And so Swagger complained about the missing version number in the JSON/YAML.

I removed this postConstruct section and moved the call of `load()` into the two getters:

```java
 /**
     * @return the versionString
     */
    public String getVersionString() {
        load();
        return versionString;
    }

    /**
     * @return the versionDate
     */
    public String getVersionDate() {
        load();
        return versionDate;
    }
```
So now we are sure that this is called every time if `versionString` or `versionDate` are null :)

Reply:
OK that's fine.  I probably would have preferred a change to `Tomcat8ApiManagerApplication` instead:

```java
Version version = new Version();
version.postConstruct();
beanConfig.setVersion(version.getVersionString());
```

But honestly it's not a huge deal.

Reply:
Just out of interest why would you prefer the other changes? :)

Reply:
It was nice having the `Version` as a CDI bean.  I'm not a huge fan of lazy loading as part of the getters.  It's error prone (e.g. if we added more properties to the class we need to remember to call `load()` in every getter) and typically not best-practice IMO.  Sometimes it's necessary.

Another valid approach would be to make `Version` a singleton.

But again, I don't feel strongly enough about it to ask you to change it. :)

Reply:
Okay, that sounds good :) 
But it is good to know what was your idea about this. Maybe we can refactor this one time in the future, but unfortunately, our resources are also limited.